[18.0][MIG] sale_import_base

rvalyi opened the pull request:
Migration of sale_import_base from 16.0 to 18.0.

### Changes
- Adapted payment provider handling for Odoo 18.0 payment architecture (offline operation, journal validation, company-aware search)
- Refactored test payment setup for new payment.method / account.payment.method relationship
- Adapted views (tree→list), Pydantic model_dump, and other v18 framework changes
- Removed deprecated pricelist_id demo references
- Used Command.link/set instead of legacy tuple notation

### Dependencies
- Depends on sale_channel_partner (already migrated to 18.0)

### Testing
- All 30 tests passing with zero warnings
- Pre-commit passing

kevinkhao committed:
[ADD] sale_import_base, sale_import_delivery_carrier, sale_import_rest

kevinkhao committed:
[FIX] Remove batch run test

kevinkhao committed:
[IMP] rename to respect OCA variable

kevinkhao committed:
[IMP] Improve tests

kevinkhao committed:
[IMP] Akretion copyrights

kevinkhao committed:
[IMP] remove _si prefixes on processor component

kevinkhao committed:
[IMP] rename json_import to data

kevinkhao committed:
[REF] Simplify sale importer component, cleanup tests

kevinkhao committed:
[FIX] Inheritance for datamodels, delivery carrier logic and tests

kevinkhao committed:
[IMP] Create sale order before sale order lines due to onchange invalidating cache
Test improvements on delivery carrier line generation

kevinkhao committed:
Update sale_import_base/components/importer_sale_channel.py

Co-authored-by: beau sebastien <sebastien.beau@akretion.com>

kevinkhao committed:
Update sale_import_base/tests/test_sale_order_import.py

Co-authored-by: beau sebastien <sebastien.beau@akretion.com>

kevinkhao committed:
Add tests, improve code logic, move validator from datamodel to importer

kevinkhao committed:
Tweaks to invoice, add force number and force date to sale order

sebastienbeau committed:
[IMP] simplify the code

- add a specific validator for customer (as required field are not the
same)
- remove currency as it's deduce from pricelist
- remove useless validate_misc due to previous change

sebastienbeau committed:
[REF] rename transaction_id to acquirer_reference and map it correctly

sebastienbeau committed:
[FIX] acquirer_reference should not be required (check, cash...)

sebastienbeau committed:
[IMP] we should pass the name of the sale order

sebastienbeau committed:
[IMP] make external amount optional as we may no check them

sebastienbeau committed:
[IMP] add possibility to set the phone and mobile

sebastienbeau committed:
[IMP] make the pricelist optionnal by reading it from the channel

sebastienbeau committed:
[FIX] add missing partner on transaction

kevinkhao committed:
[IMP] Modifications related to changing "required" attribute on datamodels: includes SO generation, tests, components, demo data
Add tests for cases where data content is full/minimal/mixed
Fix state_code: search also according to country_id
Fix state_code, country_code validation and search

kevinkhao committed:
[IMP] Add date to sale order

kevinkhao committed:
[FIX] Revert required attr on amounts

kevinkhao committed:
[REF] On import, remove pre-emptive validation and error list, instead raise on first error
[IMP] Validate payment currency as it should match Sale Order's pricelist's currency

kevinkhao committed:
[IMP] Improve UI for sale channel form view

kevinkhao committed:
[14.0][MIG] Migration to 14.0

kevinkhao committed:
[IMP] run pre-commit

kevinkhao committed:
[14.0] 14.0 Branch: Initial commit with no installable modules

sebastienbeau committed:
[IMP] add the possibility to confirm order and generate invoice

kevinkhao committed:
[IMP] Improve pluggability of ImporterSaleChannel component

kevinkhao committed:
[FIX] Fix partner versioning using alternate way of dealing with addresses

kevinkhao committed:
[IMP] Add types to addresses

kevinkhao committed:
[IMP] Create addresses in inactive state, add tests

kevinkhao committed:
[IMP] Add notion of choosing either external identifier or Odoo native numbering
Minor typo fix
Minor test refactor

sebastienbeau committed:
[IMP] improve error message

sebastienbeau committed:
[FIX] fix missing date

florian-dacosta committed:
[FIX] Avoid sale order duplicates in case a same chunk is played mutliple times

florian-dacosta committed:
[FIX] Tests after prevening SO duplicates

florian-dacosta committed:
Make exception rules unactived by default + add the no update flag to make it customizable

florian-dacosta committed:
[FIX] tests and ci

github-grap-bot committed:
[UPD] README.rst

github-grap-bot committed:
sale_import_base 14.0.1.1.0

sebastienbeau committed:
sale_import_base: refactor and migration to V16

sebastienbeau committed:
sale_import_base: use extendable_pydantic

florian-dacosta committed:
[FIX] Adapt schema to pydantic v2

florian-dacosta committed:
[FIX] tests

florian-dacosta committed:
[FIX] replace deprecated dict method by model_dump

florian-dacosta committed:
[FIX] Make main SaleOrder pydantic class extendable

florian-dacosta committed:
[FIX] warnings : add dummy security rule for sale.channel.importer + make sale.import.service.sale an Abstract Model

bealdav committed:
IMP sale_imp_base: remove product_unique_code

clementmbr committed:
[IMP] add crm_team_id and methods to ease inherit sale.channel.importer

clementmbr committed:
[IMP] search product related to its company_id

clementmbr committed:
[IMP] search product with no company_id

florian-dacosta committed:
[IMP] Add hook to allow to disable the invoice and delivery address merge and always have 2 distinct addresses
In case you want to update the delivery adresse after the creation of the sale order, it is easier to already have distinct addresses between delivery and invoicing.
Else one should create a new address, update the sale order, update the delivery order...
In the future, it  could become an option on channel or company instead of this hook that require an override to change the behavior

Commit:
[FIX] pre-commit clean for OCA

Commit:
[FIX] test mock and queue.job.chunk creation

florian-dacosta committed:
[REF] Refactoer queue_job_chunk and include it in sale_import_base
queue.job.chunk is renamed sale.import.payload

florian-dacosta committed:
[IMP] vacuum sale payload after 365 days by default + add chatter to log errors

Commit:
[IMP] add archive_addresses field

florian-dacosta committed:
[FIX] Remove unused invoice data

florian-dacosta committed:
[IMP] Add hook for searching the sale order line product

Commit:
[UPD] Update sale_import_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (78 of 78 strings)

Translation: sale-channel-16.0/sale-channel-16.0-sale_import_base
Translate-URL: https://translation.odoo-community.org/projects/sale-channel-16-0/sale-channel-16-0-sale_import_base/it/

florian-dacosta committed:
[FIX] sale_import_base: Syntax of unique constraint

Commit:
[UPD] Update sale_import_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-channel-16.0/sale-channel-16.0-sale_import_base
Translate-URL: https://translation.odoo-community.org/projects/sale-channel-16-0/sale-channel-16-0-sale_import_base/

rvalyi committed:
[IMP] sale_import_base: pre-commit auto fixes

rvalyi committed:
[MIG] sale_import_base: Migration to 18.0

rvalyi commented:
/ocabot migration sale_import_base

rvalyi commented:
/ocabot migration sale_import_base

florian-dacosta commented:
@rvalyi Can you rebase/ remove the test-requirement commit as the dependency has been merged?
Is the PR ready (I see it is draft and last commit mention a WIP) ?

rvalyi commented:
> @rvalyi Can you rebase/ remove the test-requirement commit as the dependency has been merged? Is the PR ready (I see it is draft and last commit mention a WIP) ?

I rebased and adjusted a few details. It's now ready for review. It's allowing the sale_import_rest PR https://github.com/OCA/sale-channel/pull/44 to pass all tests.
cc @sebastienbeau @bguillot

florian-dacosta reviewed (changes requested):
Tested a bit a runboat : 
The form view of the sale.payload is not working : 

<img width="1852" height="454" alt="Image" src="https://github.com/user-attachments/assets/65165a96-e5a8-4292-87f3-aaf2ec176d58" />

In v16 we have something like that : 

<img width="1852" height="604" alt="Image" src="https://github.com/user-attachments/assets/5f60a188-0803-4154-82b2-72664c01a1d0" />

Otherwise, it seems to work as expected

rvalyi commented:
I'll fix that tomorrow sorry.

florian-dacosta committed:
[FIX] sale payload form view

dnplkndll reviewed (approved):
LGTM

hparfr commented on sale_import_base/tests/test_sale_order_import.py:
```diff
@@ -58,10 +58,8 @@ def test_name_clientref(self):
-            self.get_created_sales().name[0], "S"
-        )  # native name is S + padding length 5
-        self.assertEqual(len(self.get_created_sales().name), 6)
+        # native name is S + padding length 5 (e.g. S00001)
```
padding not tested.

hparfr commented on sale_import_base/migrations/16.0.2.0.0/post-migration.py:
```diff
@@ -0,0 +1,9 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
to remove
